src/main/kotlin/cn/varsa/idea/pde/partial/core/manifest/ManifestIndex.kt

```diff
@@ -0,0 +1,59 @@
+package cn.varsa.idea.pde.partial.core.manifest
+
+import cn.varsa.idea.pde.partial.common.*
+import cn.varsa.idea.pde.partial.common.manifest.*
+import com.intellij.openapi.diagnostic.*
+import com.intellij.openapi.progress.*
+import com.intellij.util.indexing.*
+import com.intellij.util.io.*
+import org.jetbrains.kotlin.idea.core.util.*
+import org.jetbrains.lang.manifest.*
+import java.io.*
+import java.util.jar.*
+
+object ManifestIndex : SingleEntryFileBasedIndexExtension<BundleManifest>() {
+  val KEY: ID<Int, BundleManifest> = ID.create(ManifestIndex::class.java.canonicalName)
+  private val LOG = Logger.getInstance(ManifestIndex::class.java)
+
+  override fun getName() = KEY
+  override fun dependsOnFileContent() = true
+  override fun getVersion() = 2
+  override fun getIndexer() = INDEXER
+  override fun getValueExternalizer() = Externalizer
+  override fun getInputFilter() = FileBasedIndex.InputFilter { virtualFile ->
+    virtualFile.fileType is ManifestFileType && virtualFile.name == Constants.Partial.File.MANIFEST_MF
+  }
+
+  private val INDEXER = object : SingleEntryIndexer<BundleManifest>(false) {
+    override fun computeValue(inputData: FileContent): BundleManifest? = try {
+      if (inputData.fileType is ManifestFileType) {
+        val manifest = Manifest(ByteArrayInputStream(inputData.content))
+        BundleManifest(manifest)
```

Comment:
It is now straightforward to index MANIFEST.MF as a BundleManifest object.
Is it necessary not to build a BundleManifest object, but to index the Header directly? Similar to com.intellij.xml.index.XmlTagNamesIndex, what we need is not this object, but its Header and its value, so that the subsequent plugins.xml and Extensions Point Schema can also be used Build the same index.